org.eclipse.lyo.oslc4j.provider.jena/src/main/java/org/eclipse/lyo/oslc4j/provider/jena/JenaModelHelper.java

```diff
@@ -489,16 +492,38 @@ public static void skolemize(final Model m, Function<BlankNodeId, String> skolem
 		}
 	}
 
+        private static Map<String, Class> getRdfTypeClasses(Package pkg) {
+            String name;
+            OslcName oslcName;
+            OslcNamespace oslcNs;
+            Set<Class<?>> modelClasses;
+            Map<String, Class> rdfTypeClasses = new HashMap<>();
+            try {
+                modelClasses = ClassScanner.getClassesOnPackage(pkg.getName());
+                for (Class<?> modelClass : modelClasses) {
+                    oslcNs = modelClass.getAnnotation(OslcNamespace.class);
+                    if (oslcNs != null) {
+                        oslcName = modelClass.getAnnotation(OslcName.class);
+                        name = (oslcName == null ? modelClass.getSimpleName() : oslcName.value());
+                        rdfTypeClasses.put(oslcNs.value() + name, modelClass);
+                    }
+                }
+            } catch(ClassNotFoundException | IOException ex) {
+                logger.warn("Could not map RDF types to model classes", ex);
+            }
+            return rdfTypeClasses;
+        }
+
 	private static List<Object> createObjectResultList(final Class<?> beanClass,
 			List<Object> results, List<Resource> listSubjects)
 			throws IllegalAccessException, InstantiationException,
 			DatatypeConfigurationException, InvocationTargetException,
 			OslcCoreApplicationException, URISyntaxException,
 			NoSuchMethodException {
 		if (null != listSubjects) {
-
+                        Map<String, Class> rdfTypeClasses = getRdfTypeClasses(beanClass.getPackage());
```

Comment:
but here you are assuming the subclasses are in the same package? but we could have a case where such classes are in the `.impl` subpackage.

Reply:
Yes, you're right. I initial thought in defining extra packages where you want to scan for types, maybe using the class OSLC4JUtils, but I wanted to wait this discussion to see whether it was the best idea or not.